OpenRA.Mods.Common/Traits/Render/WithMakeAnimation.cs

```diff
@@ -15,40 +15,62 @@
 namespace OpenRA.Mods.Common.Traits.Render
 {
 	[Desc("Replaces the sprite during construction.")]
-	public class WithMakeAnimationInfo : ITraitInfo, Requires<WithSpriteBodyInfo>
+	public class WithMakeAnimationInfo : ITraitInfo, Requires<WithSpriteBodyInfo>, Requires<BuildingInfo>
 	{
 		[Desc("Sequence name to use")]
 		[SequenceReference] public readonly string Sequence = "make";
 
+		[GrantedConditionReference]
+		[Desc("The condition to grant to self while the make animation is playing.")]
+		public readonly string Condition = null;
+
 		public object Create(ActorInitializer init) { return new WithMakeAnimation(init, this); }
 	}
 
-	public class WithMakeAnimation
+	public class WithMakeAnimation : INotifyCreated
 	{
 		readonly WithMakeAnimationInfo info;
 		readonly WithSpriteBody wsb;
+		int token = ConditionManager.InvalidConditionToken;
 
 		public WithMakeAnimation(ActorInitializer init, WithMakeAnimationInfo info)
 		{
 			this.info = info;
 			var self = init.Self;
 			wsb = self.Trait<WithSpriteBody>();
+		}
 
+		void INotifyCreated.Created(Actor self)
+		{
 			var building = self.TraitOrDefault<Building>();
 			if (building != null && !building.SkipMakeAnimation)
 			{
+				var conditionManager = self.TraitOrDefault<ConditionManager>();
+				if (conditionManager != null && !string.IsNullOrEmpty(info.Condition) && token == ConditionManager.InvalidConditionToken)
+					token = conditionManager.GrantCondition(self, info.Condition);
+
 				wsb.PlayCustomAnimation(self, info.Sequence, () =>
 				{
+					if (conditionManager != null && token != ConditionManager.InvalidConditionToken)
```

Comment:
`conditionManager != null` should be true if `token != ConditionManager.InvalidConditionToken` is true.
However, that is too minor of a detail to hold this PR off longer.